service/history/workflow/context.go

```diff
@@ -873,11 +875,12 @@ func (c *ContextImpl) ReapplyEvents(
 	return err
 }
 
-func (c *ContextImpl) UpdateRegistry(ctx context.Context) update.Registry {
+func (c *ContextImpl) UpdateRegistry(ctx context.Context, ms MutableState) update.Registry {
 	if c.updateRegistry == nil {
-		nsIDStr := c.MutableState.GetNamespaceEntry().ID().String()
+		// NOTE: cannot use Context's MutableState, as it might not be loaded - use the parameter instead.
+		nsIDStr := ms.GetNamespaceEntry().ID().String()
 		c.updateRegistry = update.NewRegistry(
-			func() update.Store { return c.MutableState },
+			func() update.Store { return ms },
```

Comment:
Not related to this PR. 

My understanding is with the change the loaded ms still may not be consistent with the currently cached update registry. And we need to resolve the lifecycle issue between those two components. Can you maybe add a TODO here? I think @alexshtin is working on this issue.

Reply:
Right; this PR shouldn't change any of that behavior. For the existing use cases, the source of the MS is still the same (ie the `workflow.Context`). But for the case where the Context hasn't been persisted yet, it'll use the pass-in, in-memory MS instead of crashing due to a `nil` pointer exception.

Reply:
Talked to Alex and changed the approach.